crowbar_framework/app/models/crowbar_service.rb

```diff
@@ -349,6 +349,10 @@ def prepare_nodes_for_os_upgrade
       role = proposal.role
       proposal.raw_data["attributes"]["pacemaker"]["clone_stateless_services"] = false
       proposal.raw_data["attributes"]["pacemaker"]["drbd"]["enabled"] = false
+      # Save the original value of clone_stateless_services, we need it for determining
+      # some actions from recipes.
+      role.default_attributes["pacemaker"]["clone_stateless_services_orig"] =
+        role.default_attributes["pacemaker"]["clone_stateless_services"]
```

Comment:
We should delete  ` role.default_attributes["pacemaker"]["clone_stateless_services_orig"]` at some point again.

The earliest opportunity would be after the `commit_and_check_proposal` call on line 365. But maybe it's better to actually do it once the upgrade completed.

Reply:
Yeah, I've still not decided about the right place...

Reply:
Actually right after the `commit_and_check_proposal` might be a good idea, later we do not have any nice place where we tackle the roles again...

Reply:
update: actually I moved it to the very end of `services` method